tests/openxml/wordprocessing/test_level.py

```diff
@@ -0,0 +1,83 @@
+# coding: utf-8
+from __future__ import (
+    absolute_import,
+    print_function,
+    unicode_literals,
+)
+
+from unittest import TestCase
+
+from pydocx.openxml.wordprocessing import (
+    Level,
+    ParagraphProperties,
+    RunProperties,
+)
+from pydocx.util.xml import parse_xml_from_string
+
+
+class LevelTestCase(TestCase):
+    def _load_from_xml(self, xml):
+        root = parse_xml_from_string(xml)
+        return Level.load(root)
+
+    def test_ilvl_mapped_to_level_id_attribute(self):
+        xml = b'''
+            <lvl ilvl="100">
+            </lvl>
+        '''
+        level = self._load_from_xml(xml)
+        self.assertEqual(level.level_id, "100")
+
+    def test_starting_position_attribute(self):
+        xml = b'''
+            <lvl>
+                <start val="50" />
+            </lvl>
+        '''
+        level = self._load_from_xml(xml)
+        self.assertEqual(level.start, "50")
+
+    def test_num_format_attribute(self):
+        xml = b'''
+            <lvl>
+                <numFmt val="decimal" />
+            </lvl>
+        '''
+        level = self._load_from_xml(xml)
+        self.assertEqual(level.num_format, "decimal")
+
+    def test_restart_attribute(self):
+        xml = b'''
+            <lvl>
+                <lvlRestart val="1" />
+            </lvl>
+        '''
+        level = self._load_from_xml(xml)
+        self.assertEqual(level.restart, "1")
+
+    def test_associated_paragraph_style_attribute(self):
+        xml = b'''
+            <lvl>
+                <pStyle val="normal" />
+            </lvl>
+        '''
+        level = self._load_from_xml(xml)
+        self.assertEqual(level.paragraph_style, "normal")
+
+    def test_run_properties_child(self):
+        xml = b'''
+            <lvl>
+                <rPr />
+            </lvl>
+        '''
+        level = self._load_from_xml(xml)
+        self.assertEqual(level.run_properties.__class__, RunProperties)
```

Comment:
The first resource makes a solid case that `isinstance` is preferred to `__class__` *when you are writing control flow logic*. Totally agree. I'm not doing that here. 

Also, as confirmed by the first resource, `isinstance` would return true if the class is a sub-class. In the model implementation schema, I feel that would be a bug. The class should not be a sub-class (of `RunProperties` in this case), but the exact class itself.

The PEP8 resource doesn't mention `__class__` at all. Instead, it says you should not use `type` in a conditional. I'm not using type, and I don't have a conditional. It doesn't say that `__class__` is to be avoided.

Just so we're on the same page, the alternative we're discussing is the following:

```python
assert isinstance(level.run_properties, RunProperties)
```

Downsides:

* When the assertion fails, there's no visibility to what the actual class is, just what it is not
* Captures sub-classes, which may not be desired (I feel they are not)

Upsides:

* Avoids having to access a dunder property
* This is how type checking is preferred for control flow

Thoughts?